### ISSUE TITLE
API - Add a term that waits for a table to be ready

### ISSUE THREAD

**Issue by neumino** [rethinkdb/rethinkdb#2259]
`tableList` can return a table that is not yet available, and we currently don't have a way to wait for the table to be ready.

It happens when in thinky I create a Model (that creates the table) and a relation (that creates an index on the table).
This was the real reason behind #2248 -- `tableCreate` returns only when a table is available.

This script quickly ends up throwing an error.

```js
var r = require('rethinkdb');

var s8 = function() {
    return Math.floor((1+Math.random())*0x100000).toString(16).substring(1)+Math.floor((1+Math.random())*0x100000).toString(16).substring(1);
}
var i = 0;
r.connect({port: 42935}, function(err, conn1) {
r.connect({port: 42935}, function(err, conn2) {
    var run = function() {
        var name = s8();
        var count = 0;
        r.tableCreate(name).do(function(result) {
            return r.table(name).indexCreate("foo").do(function(result) {
                return r.table(name).indexWait("foo");
            });
        }).run(conn1, function(err, result) {
            if (err) throw err;
            count++;
            if (count === 2) {
                console.log(i++);
                run();
            }
        });

        r.branch(
            r.tableList().contains(name),
            r.table(name).indexCreate("bar").do(function(result) {
                return r.table(name).indexWait("bar");
            }),
            r.expr("Nothing to do")
        ).run(conn2, function(err, result) {
            if (err) throw err;
            count++;
            if (count === 2) {
                console.log(i++);
                run();
            }
        });

    };
    run();

});
});
```

**Comment by Tryneus** [comment]
This is in review 2444.  The only difference between the implementation and the description above is that `all_replicas_up` is now `all_replicas_ready` - I believe that was originally a typo anyway, the implementation matches the `table_status` row format.

**Comment by Tryneus** [comment]
Support for the `wait_for` and `timeout` optargs has been approved and merged to `next` as of commit 4a25535.  Will be in release 1.16.  This kills the issue.
